examples/rag_report_demo_why_some_customers_donot_like_pizza.sql

```diff
@@ -0,0 +1,187 @@
+select pg_catalog.current_setting('ai.openai_api_key', true) as api_key;
+
+-- check openai connection
+select * from openai_list_models()  olm
+where id in ('gpt-4o-mini','text-embedding-3-small');
+
+-- check the gpt-4o-mini , text-embedding-3-smal work well.
+with 
+tmp as (select 'what is your name?' as message )
+, test_result as (
+	select
+		message, 
+		openai_chat_complete(
+			'gpt-4o-mini',
+			jsonb_build_array(
+				jsonb_build_object('role', 'user', 'content', message)
+			)) as chat, 
+		openai_embed('text-embedding-3-small',message) as embed
+	from tmp)
+select * from test_result;
+
+
+---- step 2 create customer feedback data table and insert the demo data.
+--- the original customer feedback message to food 
+CREATE TABLE public.t_demo_text_v1 (
+id bigserial NOT NULL,
+title text NOT NULL,
+url text NULL,
+customer_message text NULL,
+text_length INTEGER GENERATED ALWAYS AS (LENGTH(customer_message)) stored,
+CONSTRAINT t_demo_text_v1_pkey PRIMARY KEY (id)
+);
+--- insert data of t_demo_text_v1
+INSERT INTO public.t_demo_text_v1 (title,url,customer_message) VALUES
+	 ('Review pizza',NULL,'The best pizza I''ve ever eaten. The sauce was so tangy!'),
+	 ('Review pizza',NULL,'The pizza was disgusting. I think the pepperoni was made from rats.'),
+	 ('Review pizza',NULL,'I ordered a hot-dog and was given a pizza, but I ate it anyway.'),
+	 ('Review pizza',NULL,'I hate pineapple on pizza. It is a disgrace. Somehow, it worked well on this izza though.'),
+	 ('Review pizza',NULL,'I ate 11 slices and threw up. The pizza was tasty in both directions.');
+
+-- the data of embedding result
+CREATE TABLE public.t_embeddings_v1 (
+id bigserial NOT NULL,
+text_id text NOT NULL, 
+text_content text NOT NULL, -- it is same as t_demo_text_v1
+model_name text NOT NULL,
+ntoken int4 NULL,
+nlength int4 NULL,
+embedding public.vector NOT NULL,
+CONSTRAINT t_embeddings_v1_pkey PRIMARY KEY (id)
+);
+
+
+----------------------------------------------------------------------------------
+-- step 3. Convert the original customer message to food into vector
+----------------------------------------------------------------------------------
+
+-- get embedding result and insert into the embedding table 
+with tmp as (
+select
+	tt.id, tt.customer_message,
+	'text-embedding-3-small'::text as model_name,
+	openai_embed('text-embedding-3-small',customer_message) as embedding
+from
+	t_demo_text_v1 as tt
+)
+insert into t_embeddings_v1 
+		(text_id, text_content, model_name, embedding )
+	select 
+		id, customer_message, model_name, embedding
+	from 
+		tmp;
+
+--- create index for query speed optimization. It is optional for small data demo 
+-- CREATE INDEX ON t_embeddings_v1 USING ivfflat (embedding vector_cosine_ops) WITH (lists='5');
+-- for unknow reason, I can not build index this time. It report Error: column does not have dimensions. 
+-- as small data, skip the index part at this moment
```

Comment:
https://platform.openai.com/docs/models/embeddings

`text-embedding-3-small` uses 1,536 dimensions use `vector(1536)` for the datatype in the column definition.

Reply:
good reference.